MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -3558,6 +3558,32 @@ UninstallMemoryAttributeProtocol (
   }
 }
 
+/**
+  Maps memory below 640K (legacy BIOS write-back memory) as readable, writeable, and executable.
+**/
+STATIC
+VOID
+MapLegacyBiosMemoryRWX (
+  VOID
+  )
+{
+  EFI_STATUS  Status;
+
+  // https://wiki.osdev.org/Memory_Map_(x86)
+  //
+  // Map the legacy BIOS write-back memory as RWX.
+  if (gCpu != NULL) {
+    Status = gCpu->SetMemoryAttributes (
+                     gCpu,
+                     0x0,
+                     0xa0000,
+                     0
+                     );
+  }
+
+  ASSERT_EFI_ERROR (Status);

```

Comment:
You'll need to initialize `Status` for the case when `gCpu` is null.

Reply:
Good catch -- fixed